test/integration/highlight_editor_spec.mjs

```diff
@@ -1760,4 +1760,231 @@ describe("Highlight Editor", () => {
+                ? /^e path through the program .*Hence, recording an$/s
+                : /^ path through the program .*Hence, recording an$/s
+            );
+
+          await moveInSteps(page, positionStartPage2, positionEndPage2, 20);
+          await page.mouse.up();
+
+          expect(await getSelectedText(page))
+            .withContext(`In ${browserName}, second selection`)
+            .toMatch(
+              browserName === "chrome"
+                ? /^e path through.*Hence, recording and .* tracing, and give up$/s
+                : /^ path through.*Hence, recording and .* tracing, and give $/s
+            );
+        })
+      );
+    });
+
+    it("doesn't jump when moving selection with carets", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          if (browserName !== "firefox") {
+            return;
+          }
+
+          const [initialStart, initialEnd, finalEnd] = await Promise.all([
+            getSpanRectFromText(
+              page,
+              1,
+              "(frequently executed) bytecode sequences, records"
+            ).then(middleLeftPosition),
+            getSpanRectFromText(
+              page,
+              1,
+              "(frequently executed) bytecode sequences, records"
+            ).then(middlePosition),
+            getSpanRectFromText(
+              page,
+              1,
+              "them, and compiles them to fast native code. We call such a se-"
+            ).then(belowEndPosition),
+          ]);
+
+          await page.mouse.move(initialStart.x, initialStart.y);
+          await page.mouse.down();
+          await moveInSteps(page, initialStart, initialEnd, 20);
+          await page.mouse.up();
+
+          expect(await getSelectedText(page))
+            .withContext(`In ${browserName}, first selection`)
+            .toBe("(frequently executed) byt");
+
+          const initialCaretPos = {
+            x: initialEnd.x,
+            y: initialEnd.y + 10,
+          };
+          const intermediateCaretPos = {
+            x: finalEnd.x,
+            y: finalEnd.y + 5,
+          };
+          const finalCaretPos = {
+            x: finalEnd.x + 20,
+            y: finalEnd.y + 5,
+          };
+
+          await page.mouse.move(initialCaretPos.x, initialCaretPos.y);
+          await page.mouse.down();
+          await moveInSteps(page, initialCaretPos, intermediateCaretPos, 20);
+          await page.mouse.up();
+
+          expect(await getSelectedText(page))
+            .withContext(`In ${browserName}, second selection`)
+            .toMatch(/^\(frequently .* We call such a se-$/s);
+
+          await page.mouse.down();
+          await moveInSteps(page, intermediateCaretPos, finalCaretPos, 20);
+          await page.mouse.up();
+
+          expect(await getSelectedText(page))
+            .withContext(`In ${browserName}, third selection`)
+            .toMatch(/^\(frequently .* We call such a se-$/s);
```

Comment:
It's indeed possible to click on the caret and drag it around with Puppeteer :) Before this PR, this test fails with the selection starting at the beginning of the page.